lib/rex/post/meterpreter/extensions/stdapi/sys/process_subsystem/memory.rb

```diff
@@ -139,7 +139,10 @@ def search(needles, min_search_len = 5, match_len = 500)
     request = Packet.create_request(COMMAND_ID_STDAPI_SYS_PROCESS_MEMORY_SEARCH)
 
     request.add_tlv(TLV_TYPE_PID, process.pid)
-    needles.each { | needle | request.add_tlv(TLV_TYPE_MEMORY_SEARCH_NEEDLE, needle) }
+    needles.each do | needle |
+      request.add_tlv(TLV_TYPE_MEMORY_SEARCH_NEEDLE, needle)
+      request.add_tlv(TLV_TYPE_MEMORY_SEARCH_NEEDLE_LEN, needle.length)
```

Comment:
String length != byte length 👀

```suggestion
      request.add_tlv(TLV_TYPE_MEMORY_SEARCH_NEEDLE_LEN, needle.bytesize)
```

I don't think it's required in this specific scenario, but it's a source of multiple bugs elsewhere in Metasploit


```
3.0.5 :003 > "👋".length
 => 1 
3.0.5 :004 > "👋".bytesize
 => 4 
```